test/language/expressions/class/fields-computed-name-propname-constructor.js

```diff
@@ -0,0 +1,34 @@
+// This file was procedurally generated from the following sources:
+// - src/class-fields/propname-constructor.case
+// - src/class-fields/propname-error/cls-expr-computed-name.template
+/*---
+description: Static class fields cannot have PropName 'prototype' (computed)
+esid: sec-class-definitions-static-semantics-early-errors
+features: [class-fields]
+flags: [generated]
+info: |
+    Static Semantics: PropName
+    ...
+    ComputedPropertyName : [ AssignmentExpression ]
+      Return empty.
+
+    // This test file tests the following early error:
+    ClassElement : FieldDefinition;
+      It is a Syntax Error if PropName of FieldDefinition is "constructor".
+
+    FieldDefinition : ClassElementNameInitializer
+      Return PropName of ClassElementName.
+
+    ClassElementName : PropertyName
+      Return PropName of PropertyName.
+
+    ClassElementName : PrivateName
+      Return empty.
+
+---*/
+
+
+x = constructor;
```

Comment:
Should I have this positive tests?

Reply:
Not sure, but I'm wondering if `constructor` is supposed to be a string `"constructor"`?

Reply:
this can be further tested if you add a assert.throw when you construct a new instance of `C`.

In anyway this test file is valid already for static semantics, after the suggestions from Rick are applied in this line. (`var x = "constructor";`

It's really nice to have this test case.

Btw, out of curiosity:

```js
var constructor = "lol";

class C {
  [constructor];
}
var c = new C();
assert.sameValue(c.hasOwnProperty("lol"), true);
```

Reply:
@leobalter I'm having troubling finding in the spec where/how it will error if there is a class field with the PropName "constructor" -- I think what happens is this:

You can see the making of the constructor from the constructor method starting at step 13 [here](https://tc39.github.io/proposal-class-fields/#runtime-semantics-class-definition-evaluation)
```
13. Else, let constructor be ConstructorMethod of ClassBody.
```
ConstructorMethod looks for a ClassElement with PropName "constructor" -- so the computer property name field won't surface here. The computed property name is evaluated in step 27.a.i, and then added to the instance while the instance is being constructed. 

So no error? I will check that it doesn't error when constructing a new instance, though!